Mathlib/NumberTheory/IntervalSums.lean

```diff
@@ -0,0 +1,189 @@
+intervals `Icc (-N) N`· -/
+def symmetricConditional : SummationFilter G where
+  filter := atTop.map (fun g ↦ Icc (-g) g)
+
+/-- The SummationFilter on `ℤ` corresponding to the intervals `Icc -N N`. Note that this is
+the same as the limit over open intervals `Ioo -N N` (see `symCondInt_eq_map_Ioo`). -/
+abbrev symCondInt : SummationFilter ℤ := symmetricConditional ℤ
+
+lemma symmetricConditional_eq_map_Icc :
+    (symmetricConditional G).filter = atTop.map (fun N ↦ Icc (-N) N) := rfl
+
+lemma symCondInt_eq_map_Icc_nat :
+    symCondInt.filter = atTop.map (fun N : ℕ ↦ Icc (-(N : ℤ)) N) := by
+  rw [symCondInt, symmetricConditional, ← Nat.map_cast_int_atTop]
+  rfl
+
+/-- The SummationFilter on `G` corresponding to the intervals `Ico (-N) N`. -/
+def IcoFilter : SummationFilter G where
+  filter := atTop.map (fun N ↦ Ico (-N) N)
+
+/-- The SummationFilter on `G` corresponding to the intervals `Ioc (-N) N`. -/
+def IocFilter : SummationFilter G where
+  filter := atTop.map (fun N ↦ Ioc (-N) N)
+
+lemma symCondInt_eq_map_Ioo : symCondInt.filter = atTop.map (fun N ↦ Ioo (-N) N) := by
+  ext s
+  simp only [symmetricConditional, ← Nat.map_cast_int_atTop, Filter.mem_map, mem_atTop_sets,
+    Set.mem_preimage]
+  refine ⟨fun ⟨a, ha⟩ ↦ ⟨a + 1, fun b hb ↦ ?_⟩, fun ⟨a, ha⟩ ↦ ⟨a - 1, fun b hb ↦ ?_⟩⟩ <;>
+  [ convert ha (b - 1) (by grind) using 1; convert ha (b + 1) (by grind) using 1 ] <;>
+  · simp only [Finset.ext_iff, mem_Ioo, mem_Icc]
+    grind
+
+variable [(atTop : Filter G).NeBot]
+
+instance : (symmetricConditional G).NeBot where
+  ne_bot := by simp [symmetricConditional, Filter.NeBot.map]
+
+instance : (IcoFilter G).NeBot where
+  ne_bot := by simp [IcoFilter, Filter.NeBot.map]
+
+instance : (IocFilter G).NeBot where
+  ne_bot := by simp [IocFilter, Filter.NeBot.map]
+
+section LeAtTop
+
+variable {G : Type*} [AddCommGroup G] [PartialOrder G] [IsOrderedAddMonoid G] [LocallyFiniteOrder G]
+
+lemma symmetricConditional_le_Conditional :
+    (symmetricConditional G).filter ≤ (conditional G).filter :=
+  Filter.map_mono (tendsto_neg_atTop_atBot.prodMk tendsto_id)
+
+instance : (symmetricConditional G).LeAtTop where
+  le_atTop := le_trans symmetricConditional_le_Conditional (conditional G).le_atTop
+
+instance : (IcoFilter ℤ).LeAtTop where
+  le_atTop := by
+    rw [IcoFilter, map_le_iff_le_comap, ← @tendsto_iff_comap]
+    exact tendsto_Ico_atTop_atTop
+
+instance : (IocFilter ℤ).LeAtTop where
+  le_atTop := by
+    rw [IocFilter, map_le_iff_le_comap, ← @tendsto_iff_comap]
+    exact tendsto_Ioc_atTop_atTop
+
+end LeAtTop
+
+end SummationFilter
+
+section comparisons
+
+variable {α : Type*} {f : ℤ → α} [CommGroup α] [TopologicalSpace α] [ContinuousMul α]
+
+@[to_additive]
+lemma multipliable_IcoFilter_of_multiplible_symCondInt
+    (hf : Multipliable f symCondInt) (hf2 : Tendsto (fun N : ℕ ↦ (f N)⁻¹) atTop (𝓝 1)) :
+    Multipliable f (IcoFilter ℤ) := by
+  have := (hf.hasProd)
```

Comment:
Maybe an "unnecessary-brackets linter" is the project I need to teach me metaprogramming.